docs/antrea-network-policy.md

```diff
@@ -253,8 +253,8 @@ default tier i.e. the "application" Tier.
 **ingress**: Each ClusterNetworkPolicy may consist of zero or more ordered
 set of ingress rules. Each rule, depending on the `action` field of the rule,
 allows or drops traffic which matches both the `from` and `ports` sections.
-Also, each rule has an optional `name` field, which is unique within the policy 
-describing the intention of this rule.
+Also, each rule has an optional `name` field, which should be unique within 
```

Comment:
Question - why rule name must be unique? I think it does really matter much even two rules have the same name (thought some features use rule name to identify a rule might get incorrect results).
@abhiraut

Reply:
i think this requirement came from @tnqn so we could use rule name to identify a rule to report stats per rule

Reply:
I guess you first need a way to indicate a name is auto-generated?

Reply:
Normally names of subresources of a resource are required to be unique within the resource. It helps status report for subresources and also makes the field "name" meaningful.
https://github.com/kubernetes/kubernetes/blob/096819c9635035abb4c4ec2bcaf4db8c35250cb7/pkg/apis/core/types.go#L48
https://github.com/kubernetes/kubernetes/blob/096819c9635035abb4c4ec2bcaf4db8c35250cb7/pkg/apis/core/types.go#L1752
https://github.com/kubernetes/kubernetes/blob/096819c9635035abb4c4ec2bcaf4db8c35250cb7/pkg/apis/core/types.go#L2095
https://github.com/kubernetes/kubernetes/blob/096819c9635035abb4c4ec2bcaf4db8c35250cb7/pkg/apis/core/types.go#L3707